src/events/guild-create.ts

```diff
@@ -1,14 +1,25 @@
-import {Guild, TextChannel, Message, MessageReaction, User} from 'discord.js';
+import {Guild, TextChannel, Message, MessageReaction, User, ApplicationCommandData,
+} from 'discord.js';
 import emoji from 'node-emoji';
 import pEvent from 'p-event';
 import {Settings} from '../models/index.js';
 import {chunk} from '../utils/arrays.js';
+import container from '../inversify.config.js';
+import Command from '../commands';
+import {TYPES} from '../types.js';
 
 const DEFAULT_PREFIX = '!';
 
 export default async (guild: Guild): Promise<void> => {
   await Settings.upsert({guildId: guild.id, prefix: DEFAULT_PREFIX});
 
+  // Setup slash commands
+  const commands: ApplicationCommandData[] = container.getAll<Command>(TYPES.Command)
```

Comment:
As per a note in the draft PR, I was planning on registering commands bot-wide in production mode and guild-wide in development mode.

So I think this needs to be wrapped in a check to see if we're in dev mode, and if you have the time it'd be great if you could add bot-wide registering in bot.ts for production mode.

You can check the current environment with `process.env.NODE_ENV===production` (probably should be added to `services/config.ts`). The Dockerfile will also need to be updated to set the environment to production, and it'd be nice if Muse said whether it was registering bot-wide or guild-wide commands when starting up.

Reply:
Ok, I'm going to change it as soon as I can.
I also have to look up how to set commands bot-wide, do you already have looked up how to do it?

Reply:
https://discordjs.guide/interactions/registering-slash-commands.html#global-commands